librdmacm/rsocket.c

```diff
@@ -3166,6 +3172,12 @@ int rpoll(struct pollfd *fds, nfds_t nfds, int timeout)
 		if (ret)
 			break;
 
+		if (timeout >= 0) {
+			timeout -= (int) ((rs_time_us() - start_time) / 1000);
+			if (timeout <= 0)
+				return 0;
```

Comment:
Shouldn't we at least do a poll with a timeout of 0 here?

Reply:
The functional equivalent of poll() with a timeout of 0 has already been done (rs_poll_arm).  Here we're simply calculating the new wait timeout, and if no additional waiting is needed, we're done.